### DISCUSSION TITLE
CodeQL Not Working with Lombok

### DISCUSSION THREAD

**Discussion by mh-emartin** [community/community#31524] [Code Security] [answered]
Hello,
It appears that my (Java) CodeQL scans are not working because of `Lombok`, per this warning: `Warning: Use of Lombok detected. Skipping file...`

I am attempting to de-Lombok the code prior to scanning, but I'm getting this error from Gradle:
```
FAILURE: Build failed with an exception.

* What went wrong:
failed to access class com.google.common.collect.AbstractMultiset$EntrySet from class com.google.common.collect.AbstractMultiset (com.google.common.collect.AbstractMultiset$EntrySet is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @64a294a6; com.google.common.collect.AbstractMultiset is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @7e0b37bc)
```
This seems like a Gradle script issue, but I'm having trouble finding answers.
- Does anyone have a working example of this for CodeQL scans using Gradle?
- Is there another approach other than de-Lomboking?

**Comment by aeisenberg** [comment]
Thanks for your question. I've asked the team supporting the Java extractor to provide some help.

**Comment by yo-h** [comment, marked as answer]
Lombok is not supported by CodeQL. A potential workaround using delombok is described at https://github.com/github/codeql/issues/4984#issuecomment-763440211.
